File: angular/dashboard/src/app/features/product-variants/models/product-variant-list-item.ts
export interface ProductVariantListItem {
  readonly id: string;
  readonly productId: number;
  readonly variantId: number;
  readonly displayName: string;
  readonly sku: string;
  readonly barcode: string;
  readonly pendingSync: boolean;
  readonly failedSyncAttempts: number;
  readonly active: boolean;
  readonly updatedOnUtc: string;
}

File: dotnet/src/Web.Api/Features/ProductVariants/GetProductVariants/ProductVariantListItem.cs
using System.Linq.Expressions;
using Infrastructure.Database.Entities;

namespace Web.Api.Features.ProductVariants.GetProductVariants;

public readonly record struct ProductVariantListItem(
    Guid Id,
    long ProductId,
    long VariantId,
    string DisplayName,
    string Sku,
    string Barcode,
    bool PendingSync,
    int FailedSyncAttempts,
    bool Active,
    DateTime UpdatedOnUtc)
{
    public static readonly Expression<Func<ShopifyProductVariantEntity, ProductVariantListItem>> Projection =
        entity => new ProductVariantListItem(
            entity.ShopifyProductVariantId,
            entity.ProductId,
            entity.VariantId,
            entity.DisplayName,
            entity.Sku,
            entity.Barcode,
            entity.PendingShopifySync,
            entity.FailedShopifySyncAttempts,
            entity.IsActive,
            entity.UpdatedOnUtc);
}
